packages/playout-gateway/src/atemUploader.ts

```diff
@@ -39,29 +40,44 @@ export class AtemUploadScript {
 		})
 	}
 
-	public loadFile(): Promise<Buffer> {
+	public loadFile(fileUrl: string): Promise<Buffer> {
 		return new Promise<Buffer>((resolve, reject) => {
-			fs.readFile(this.fileUrl, (e, data) => {
+			fs.readFile(fileUrl, (e, data) => {
 				consoleLog('got file')
 				if (e) reject(e)
 				else resolve(data)
 			})
 		})
 	}
+	public async loadFiles(folder: string): Promise<Buffer[]> {
+		const files = await fs.promises.readdir(folder)
+		consoleLog(files)
+		const loadBuffers = files.map((file) => fs.promises.readFile(path.join(folder, file)))
+		const buffers = await Promise.all(loadBuffers)
 
-	public checkIfFileExistsOnAtem(fileName: string, stillIndex: number): boolean {
+		consoleLog('got files')
+		return buffers
+	}
+
+	public checkIfFileOrClipExistsOnAtem(fileName: string, stillOrClipIndex: number, type: AtemMediaPoolType): boolean {
 		consoleLog('got a file')
 
-		const still = this.connection.state ? this.connection.state.media.stillPool[stillIndex] : undefined
-		if (still) {
-			consoleLog('has still')
-			if (still.isUsed) {
-				consoleLog('still is used')
-				if (fileName.length === AtemMaxFilenameLength) {
-					consoleLog('filename is max length, change detection might fail')
-				}
+		const still = this.connection.state ? this.connection.state.media.stillPool[stillOrClipIndex] : undefined
+		const clip = this.connection.state ? this.connection.state.media.clipPool[stillOrClipIndex] : undefined
+		let pool: typeof still | typeof clip
+		if (type === AtemMediaPoolType.Still) pool = still
+		if (type === AtemMediaPoolType.Clip) pool = clip
```

Comment:
Should be an `else if` here, given that the second if can never be true if the first one is